internal/pkg/cli/pipeline_delete.go

```diff
@@ -0,0 +1,228 @@
+		return nil, fmt.Errorf("read pipeline file %s: %w", workspace.PipelineFileName, err)
+	}
+	pipeline, err := manifest.UnmarshalPipeline(data)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal pipeline file %s: %w", workspace.PipelineFileName, err)
+	}
+
+	secretsmanager, err := secretsmanager.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create secrets manager: %w", err)
+	}
+
+	p := session.NewProvider()
+	defaultSession, err := p.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &deletePipelineOpts{
+		deletePipelineVars: vars,
+		prog:               termprogress.NewSpinner(),
+		secretsmanager:     secretsmanager,
+		pipelineDeployer:   cloudformation.New(defaultSession),
+		ws:                 ws,
+	}
+	opts.PipelineName = pipeline.Name
+	if secret, ok := (pipeline.Source.Properties["access_token_secret"]).(string); ok {
+		opts.PipelineSecret = secret
+	}
+
+	return opts, nil
+}
+
+// Ask prompts for fields that are required but not passed in.
+func (o *deletePipelineOpts) Ask() error {
+	if o.SkipConfirmation {
+		return nil
+	}
+
+	deleteConfirmed, err := o.prompt.Confirm(
+		fmt.Sprintf(pipelineDeleteConfirmPrompt, o.PipelineName, o.ProjectName()),
+		pipelineDeleteConfirmHelp)
+
+	if err != nil {
+		return fmt.Errorf("pipeline delete confirmation prompt: %w", err)
+	}
+
+	if !deleteConfirmed {
+		return errPipelineDeleteCancelled
+	}
+
+	return nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *deletePipelineOpts) Validate() error {
+	if o.ProjectName() == "" {
+		return errNoProjectInWorkspace
+	}
+
+	if o.PipelineName == "" {
+		return errNoPipelineInWorkspace
+	}
+
+	return nil
+}
+
+// Execute deletes the secret and pipeline stack
+func (o *deletePipelineOpts) Execute() error {
+	if err := o.deleteSecret(); err != nil {
+		return err
+	}
+
+	if err := o.deleteStack(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *deletePipelineOpts) deleteSecret() error {
+	if !o.DeleteSecret {
+		confirmDeletion, err := o.prompt.Confirm(
+			fmt.Sprintf(pipelineSecretDeleteConfirmPrompt, o.PipelineSecret, o.PipelineName),
+			pipelineDeleteSecretConfirmHelp,
+		)
+		if err != nil {
+			return fmt.Errorf("pipeline delete secret confirmation prompt: %w", err)
+		}
+
+		if !confirmDeletion {
+			log.Infof("Skipping deletion of secret %s.\n", o.PipelineSecret)
+			return nil
+		}
+	}
+
+	if err := o.secretsmanager.DeleteSecret(o.PipelineSecret); err != nil {
+		return err
+	}
+
+	log.Successf("Deleted secret %s.\n", o.PipelineSecret)
+
+	return nil
+}
+
+func (o *deletePipelineOpts) stackName() string {
+	return o.ProjectName() + "-" + o.PipelineName // based on stack.pipelineStackConfig.StackName()
+}
+
+func (o *deletePipelineOpts) deleteStack() error {
+	o.prog.Start(fmt.Sprintf(fmtDeletePipelineStart, o.PipelineName, o.ProjectName()))
+	stackName := o.stackName()
+
+	if err := o.pipelineDeployer.DeletePipeline(stackName); err != nil {
```

Comment:
What do you think of:
```suggestion
	if err := o.pipelineDeployer.DeletePipeline(o.ProjectName(), o.PipelineName); err != nil {
```
Right now `DeletePipeline` acts as a generic CFN stack delete, this would push the logic of the stack name generation to that method instead.

Reply:
I assumed that different resources would have different stackname schemas so wanted to keep the generation of that name out of the logic of the deployer's methods. I had also done it this way bc for a hot minute it made testing easier, but that may not actually be true anymore 😅